test/net/sourceforge/kolmafia/textui/command/MirrorCommandTest.java

```diff
@@ -0,0 +1,177 @@
+package net.sourceforge.kolmafia.textui.command;
+
+import static internal.helpers.Player.withSkill;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import internal.helpers.Cleanups;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.utilities.FileUtilities;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class MirrorCommandTest extends AbstractCommandTestBase {
+
+  public MirrorCommandTest() {
+    this.command = "mirror";
+  }
+
+  @BeforeEach
+  public void stopMirrorLogging() {
+    execute("");
+  }
+
+  private String getMirrorLog(String fileName) {
+    StringBuilder builder = new StringBuilder();
+    try (BufferedReader reader =
+        FileUtilities.getReader(new File(KoLConstants.ROOT_LOCATION, "chats/" + fileName))) {
+      String line;
+      // Read the full file to a string
+      while ((line = FileUtilities.readLine(reader)) != null) {
+        if (builder.length() > 0) {
+          builder.append("\n");
+        }
+        builder.append(line);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return builder.toString();
+  }
+
+  @Test
+  public void testMirrorWritesHTML() {
+    // Open the mirror
+    execute("chats/test_writes_html.txt");
+    // When `> ` is used as a prefix, the RequestLogger will colorize it assuming its a command
+    // input but that is not true at the moment, hence plain text assertion.
+    RequestLogger.printLine("> Fake command input");
+    // Log another line, this should be raw.
+    RequestLogger.printLine("Raw Line");
+    // Close the mirror
+    execute("");
+    String mirrorOutput = getMirrorLog("test_writes_html.txt");
+    assertThat(mirrorOutput, containsString("> Fake command input"));
```

Comment:
Specifically which test?

I broke this out because the other code and our discussions there make me think we are not talking about the same thing.

The alleged bug, to be fixed, is that a message appears in the gCLI but does not appear in the mirror.  I have been unable to write a failing test for this and as a result I am inclined to believe there is no such bug.

There is a feature request that, if the text written to the gCLI contains HTML (and is rendered as such in the gCLI) then the mirror should contain the HTML as well.  Tests to exercise changes to implement that feature have been preserved in this code but they have been converted to passing tests rather than commented out.  It could have been clearer, I suppose, if they had been Disabled.  In any event once it is established that the mirror command is mirroring everything, then the HTML inclusion can be implemented and my hope is that these tests will fail as a result and need to be changed.  The tests are documenting current behavior with the expectation that changed behavior will be detected.  If that is too confusing tests can be disabled.

The big thing here is the inability to find a test that demonstrates something written to the gCLI but not the mirror.

Reply:
> It's a bug that the html is not being printed as it does in gCLI, not a feature.

Ignoring the name, the test shows that the gCLI output and the mirror output are identical and that neither contains HTML.

Since running mafia, I could not get ">" to trigger html to the gCLI I assumed your expectation otherwise was incorrect.  Maybe I was mistaken?

Reply:
Ah, I see what you mean.

I believe it's a bug in that what's written to GCLI isn't exactly what's written to mirror. 

Mirror writes exactly what the logger was given, but the logger applies html to the input based on rules which is what's displayed in cli.

So it's arguable if it's a bug fix or a feature request. I was talking under the assumption that it's a bug, because I expect to see contents identical to what's written in cgli in the mirror.

Reply:
> I believe it's a bug in that what's written to GCLI isn't exactly what's written to mirror.

We're getting closer, I think.

The original assertion was that things were being written to the gCLI but not to the mirror and in the context of the original "complaint" the possibility that the missing information was merely HTML formatting was not really considered/expected.  

My thinking right now is that I want a test that doesn't use a mirror and shows HTML written to to the gCLI.  That's probably in RequestLoggerTest.  Once we have that then we can wrap a mirror around it and see the existing behavior as a test.  I'm headed down that rabbit hole but might not come out today.